test/device/add_inlink_test.cpp

```diff
@@ -0,0 +1,63 @@
+#include <gtest/gtest.h>
+
+#include "utils.hpp"
+#include <random>
+
+namespace test {
+
+class AddInlink : public testing::Test {
+public:
+    void TearDown() override {};
+    void SetUp() override {};
+};
+
+TEST_F(AddInlink, LinkIsPresent) {
+    auto source = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+    auto dest = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+    auto link = std::make_shared<TestLink>(TestLink(source, dest));
+
+    EXPECT_EQ(dest->next_inlink(), nullptr);
+    dest->add_inlink(link);
+    EXPECT_EQ(dest->next_inlink(), link);
+}
+
+TEST_F(AddInlink, SameLinkMultipleTimes) {
+    int NUMBER_OF_SOURCES = 5;
+    int MAX_LINKS = 3;
+    int NUMBER_OF_LOOPS = 3;
+
+    auto sources = createRoutingModules(NUMBER_OF_SOURCES);
+    auto dest = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+    
+    std::random_device rd;
+    std::mt19937 gen(rd());
```

Comment:
Usage of constant seed for `std::mt19937` might be good for reproducibility:
```suggestion
    const SEED = 42;
    std::mt19937 gen(seed);
```

Reply:
Moved RANDOM_SEED to utils